src/Tes.Runner/Docker/DockerExecutor.cs

```diff
@@ -63,7 +64,8 @@ public DockerExecutor(IDockerClient dockerClient, IStreamLogReader streamLogRead
             dockerPullRetryPolicy = new RetryPolicyBuilder(Options.Create(dockerPullRetryPolicyOptions))
                 .PolicyBuilder.OpinionatedRetryPolicy(Polly.Policy
                     .Handle(IsNotAuthFailure)
-                    .Or<IOException>())
+                    .Or<IOException>()
+                    .Or<HttpRequestException>(e => e.InnerException is IOException || e.StatusCode >= HttpStatusCode.InternalServerError || e.StatusCode == HttpStatusCode.RequestTimeout))
```

Comment:
Make sense to me. It would be nice to have tests to avoid regressions going forward.

Reply:
#613